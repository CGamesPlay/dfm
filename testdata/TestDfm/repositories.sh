#!/bin/bash
# Tests for multiple repositories working correctly
set -e
. "$(dirname "$0")/../helpers.sh"

export HOME="$(pwd)/home"
export DFM_DIR="$HOME/dfmdir"

mkdir -p ~/dfmdir/one ~/dfmdir/two
echo 'one' > ~/dfmdir/one/.bashrc
echo 'two' > ~/dfmdir/two/.bashrc
echo 'one' > ~/dfmdir/one/.vimrc
echo 'two' > ~/dfmdir/two/.zshrc

dfm init --repos one,two
dfm link
[ "$(readlink ~/.bashrc)" == ~/dfmdir/two/.bashrc ] || fail 'wrong bashrc selected'
[ "$(readlink ~/.vimrc)" == ~/dfmdir/one/.vimrc ] || fail 'wrong vimrc selected'
[ "$(readlink ~/.zshrc)" == ~/dfmdir/two/.zshrc ] || fail 'wrong zshrc selected'
