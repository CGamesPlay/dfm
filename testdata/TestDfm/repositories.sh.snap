$ dfm init --repos one,two
Initialized /test/home/dfmdir as a dfm directory.
$ dfm link
two/.bashrc -> /test/home/.bashrc
one/.vimrc -> /test/home/.vimrc
two/.zshrc -> /test/home/.zshrc
